File: leet_0283_mv_zeroes_Fast_swap.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        x = 0
        i = 0
        while i < len(nums):
            if nums[i] != 0:
                if i != x:
                    nums[i], nums[x] = nums[x], nums[i]
                x += 1
            i += 1

File: leet_0283_mv_zeroes_Fast_swap.js
var moveZeroes = function(nums) {
    let imove = -1, izero = 0
    while (++imove < nums.length){
        if (nums[imove] ^ 0){
            if (imove ^ izero){
                nums[izero] = nums[imove]
                nums[imove] = 0
            }
            izero++
        }
    }
};
